test/ptc/tools/cromwell.clj

```diff
@@ -24,3 +24,33 @@
     (->> (:body response)
          (misc/parse-json-string)
          (:results))))
+
+(defn work-around-cromwell-fail-bug
+  "Wait 2 seconds and ignore up to N times a bogus failure response from
+  Cromwell for workflow ID in ENVIRONMENT.  Work around the 'sore spot'
+  reported in https://github.com/broadinstitute/cromwell/issues/2671.
+  From https://github.com/broadinstitute/wfl/blob/master/api/src/zero/service/cromwell.clj#L266"
+  [n cromwell-url id]
+  (misc/sleep-seconds 2)
+  (let [fail {"status" "fail" "message" (str "Unrecognized workflow ID: " id)}
+        {:keys [body] :as bug} (try (status cromwell-url id)
+                                    (catch Exception e (ex-data e)))]
+    (misc/trace [bug n])
+    (when (and (pos? n) bug
+               (= 404 (:status bug))
+               (= fail (json/read-str body)))
+      (recur (dec n) cromwell-url id))))
```

Comment:
Is this still happening? (Probably.)
You don't have to cite the Zero source for this.
(But thank you for the thought.)